vault/resource_kv_secret_v2.go

```diff
@@ -100,6 +102,43 @@ func kvSecretV2Resource(name string) *schema.Resource {
 				Default:     false,
 				Description: "If set to true, permanently deletes all versions for the specified key.",
 			},
+
+			consts.FieldCustomMetadata: {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Custom metadata to be set for the secret",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						consts.FieldMaxVersions: {
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Computed:    true,
+							Description: "The number of versions to keep per key.",
+						},
+						consts.FieldCASRequired: {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Computed: true,
+							Description: "If true, all keys will require the cas " +
+								"parameter to be set on all write requests.",
+						},
+						consts.FieldDeleteVersionAfter: {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Description: "If set, specifies the length of time before " +
+								"a version is deleted",
+						},
+						consts.FieldData: {
+							Type:     schema.TypeMap,
+							Optional: true,
+							Computed: true,
```

Comment:
Should this ever be a computed value? Presumably the default is empty.

Reply:
I wanted to make sure we are able to import the custom metadata and set it to the TF state for the KV secret, specifically in the cases that the secret metadata was created outside of TF and via Vault. That was my thought process around setting this field to also be possibly computed. Is that overkill/unnecessary?

Reply:
I removed the `Computed` from the individual nested fields, but still set `Computed: true` for the parent `custom_metadata` field